File: SZE/CppProgramozas/20180403/Monster.cpp
#include "Monster.hpp"

Monster::Monster(int x, int y, int speed, int HP)
    : Entity(x,y),speed(speed), HP(HP) {}

bool Monster::attack(double damage){
    HP-=damage;
    return HP<=0;
}

bool Monster::move() {
    x-=speed;
    return x<=0;
}

File: Misc/RPG_projekt/src/0170_Hero_Monster_3/Monster.java

class Monster {
    Unit base;

    public Monster(String name, int health, int damage, int defense) {
        this.base = new Unit(name,health, damage, defense);
    } 

    public int getHealth() { return this.base.getHealth(); }
    public boolean isAlive() { return this.base.isAlive(); }
    public void takeDamage(int damage) { this.base.takeDamage(damage); }
    @Override
    public String toString() { return this.base.toString(); }

    public void attack(Hero hero) {
        hero.takeDamage(this.base.damage);
    }    
}